### DISCUSSION TITLE
Suggestion on improving UI performance for large entity systems

### DISCUSSION THREAD

**Discussion by jdiegmueller** [zachowj/node-red-contrib-home-assistant-websocket#743] [Feature Request]
My Homeassistant system has roughly 3200 entities entities.  On my systems, I see a roughly 10-25 second delay (depending on which computer) any time the full entity list must be loaded in to a drop-down list in the UI.  This makes it extremely painful to come in/out of editing nodes that populate a drop-down list of all entities by default.  I realize the performance associated with loading 3200 entities in to a drop-down list isn't going to change.  Rather, my suggestion would be to avoid populating drop-down lists with all entities where possible.

Compare the experience on a first-time edit of events:state vs current_state:
* events:state's default "exact" selection type populates a drop-down list with all entities, with the first one selected.  Because of this, there is a big delay waiting for that large drop-down list to get generated. On events:state, the selection type can (optionally) be changed to "regex" or "substring", both of which do not result in a UI delay on future edits of the node.
* current_state requires an entity value, but the default value is blank and the drop-down list is not populated until there is some input to filter against.  As long as a reasonable value is input ("outside_garage", for example) the filter only returns handful of entities and is generated without delay.

If all nodes that require entities to be provided defaulted to behaving like current_state, it would be a tremendous help to users of large-entity systems.  The existing UI behaviors could be kept, but perhaps just hidden behind a selection type like events:state.

**Comment by zachowj** [comment]
I know this is an issue and have been looking into solutions. Currently plan on moving the loading of data to an ajax request instead of building it all on the loading of the editor tab.
